### ISSUE TITLE
Add support for SRTM from RasterDataSources.jl

### ISSUE THREAD

**Issue by jenkspt** [rafaqz/Rasters.jl#242]
SRTM is a new dataset in [RasterDataSources.jl](https://github.com/EcoJulia/RasterDataSources.jl). The `SRTM` type needs to be added to [src/sources/rasterdatasources.jl](https://github.com/rafaqz/Rasters.jl/blob/master/src/sources/rasterdatasources.jl) to support the following interface:

```julia
Raster(SRTM; crs=..., bounds=(...))
```
### Details:
- `getraster(SRTM, ...)` returns a matrix of tile paths, which is currently unsupported by `Raster` (?)
- Since tiles are still somewhat large (6000 x 6000 pixels), using a `BlockArray` of `DiskArray`s is an option for the `Raster` array backing. ([BlockArrays.jl](https://github.com/JuliaArrays/BlockArrays.jl))

**Comment by jenkspt** [comment]
@rafaqz

**Comment by rafaqz** [comment]
Yep. Could also be generally useful to have a way to stitch tiles together, `mosaic` is very slow compared to when you know the tiles match step size and dont overlap. 

This needs 2 parts. A way to stitch the BlockArrays, and a way to combine the dimensions. Probably `combine` already has that part solved. Other than that its just a regular `Raster`.

Then we'll need a method for RasterDataSources types that have tiles, or just this SRTM to get started. 

Do you want to have a go at a PR?

**Comment by jenkspt** [comment]
Yea I'll have a go at it. Thanks for the feedback!

**Comment by rafaqz** [comment]
Sweet! Don't hesitate to ping me on things here, or make a work in progress PR. I try to streamline things but DimensionalData.jl is inherantly pretty complicated, and you may need to understand a bit of how LookupArray and Dimensions work to get this done.

**Comment by rafaqz** [comment]
I've just registered a new version of DiskArrays.jl with lazy `cat`, so we can cat together the SRTM tiles without reading the files.
